src/app/lib/device/dlna.js

```diff
@@ -17,74 +17,124 @@
         makeID: makeID,
 
         initialize: function (attrs) {
-            this.device = attrs.device;
-            this.client = new MediaRendererClient(this.device.xml);
-            this.attributes.name = this.device.name;
-            this.attributes.address = this.device.host;
+            this.player = attrs.player;
+            this.attributes.name = this.player.name;
+            this.attributes.address = this.player.host;
         },
 
         play: function (streamModel) {
-            var metadata = {
-                title: 'Butter DLNA', // upnp-mediarendered-client object
-                type: 'video',
-                url: streamModel.get('src'),
-                protocolInfo: 'http-get:*:video/mp4:*'
-            };
-
-            if (streamModel.get('subFile')) { // inject subtitles
-                metadata.subtitlesUrl = 'http:' + metadata.url.split(':')[1] + ':9999/video.srt';
+            var url = streamModel.get('src');
+            var self = this;
+            var media;
+            var url_video = url;
+            var url_subtitle = 'http:' + url.split(':')[1] + ':9999/video.srt';
+            var metadata = null;
+            var subtitle = streamModel.get('subFile');
+            if (subtitle) {
+                media = {
+                    title: streamModel.get('title'),
```

Comment:
This will trigger an error for me. Because special characters. Try something like `Common.normalize(streamModel.get('title'))` and in common.js: 

```
Common.normalize = (function () {
    var from = "ÃÀÁÄÂÈÉËÊÌÍÏÎÒÓÖÔÙÚÜÛãàáäâèéëêìíïîòóöôùúüûÑñÇç"; 
    var to =   "AAAAAEEEEIIIIOOOOUUUUaaaaaeeeeiiiioooouuuunncc";
    var mapping = {};

    for (var i = 0, j = from.length; i < j; i++) {
        mapping[from.charAt(i)] = to.charAt(i);
    }

    return function (str) {
        var ret = [];
        for (var i = 0, j = str.length; i < j; i++) {
            var c = str.charAt(i);
            if (mapping.hasOwnProperty(str.charAt(i)))
                ret.push(mapping[c]);
            else
                ret.push(c);
        }
        return ret.join('').replace(/\W/g, ' ').replace(/\s+/g, ' ');
    }
})();
```